openreview/tools.py

```diff
@@ -351,39 +351,142 @@ def replace_members_with_ids(client, group):
     group.members = ids + emails
     client.post_group(group)
 
-def iterget(get_function, **kwargs):
-    '''
-    Iterator over a given get function from the client.
-    '''
-    done = False
-    offset = 0
-    params = {
-        'limit': 1000 if 'limit' not in kwargs else kwargs['limit']
-    }
-
-    while not done:
-        params['offset'] = offset
-
-        params.update(kwargs)
-        batch = get_function(**params)
-        offset += params['limit']
-        for obj in batch:
-            yield obj
-
-        if len(batch) < params['limit']:
-            done = True
+class iterget:
+    def __init__(self, get_function, **params):
+        self.offset = 0
+
+        self.last_batch = False
+        self.batch_finished = False
+        self.obj_index = 0
+
+        self.params = params
+        self.params.update({
+            'offset': self.offset,
+            'limit': 1000
+        })
+
+        self.get_function = get_function
+        self.current_batch = self.get_function(**self.params)
+
+    def update_batch(self):
+        self.offset += self.params['limit']
+        self.params['offset'] = self.offset
+        next_batch = self.get_function(**self.params)
+        if next_batch:
+            self.current_batch = next_batch
+        else:
+            self.current_batch = []
 
-def get_all_tags(client, invitation, limit=1000):
-    '''
-    Given an invitation, returns all Tags that respond to it, ignoring API limit.
-    '''
-    return list(iterget(client.get_tags, invitation=invitation, limit=limit))
+    def __iter__(self):
+        return self
 
-def get_all_notes(client, invitation, limit=1000):
-    '''
-    Given an invitation, returns all Notes that respond to it, ignoring API limit.
-    '''
-    return list(iterget(client.get_notes, invitation=invitation, limit=limit))
+    def __next__(self):
+        if len(self.current_batch) == 0:
+            raise StopIteration
+        else:
+            next_obj = self.current_batch[self.obj_index]
+            if (self.obj_index + 1) == len(self.current_batch):
+                self.update_batch()
+                self.obj_index = 0
+            else:
+                self.obj_index += 1
+            return next_obj
+
+    next = __next__
```

Comment:
this line is the reason I had to turn iterget into an iterator class object, instead of a generator function. in python 2, iterables use the function `next()` to get the next item in the series, while in python 3, the function is called `__next__`. There isn't a way to reassign this in a generator function, as far as I know, so I had to rewrite it as an iterator class and reassign the function like this.